python/lsst/ts/observatory/control/auxtel/atcalsys.py

```diff
@@ -0,0 +1,397 @@
+
+        try:
+            await self.electrometer.cmd_startScanDt.set_start(
+                scanDuration=duration, timeout=duration + self.long_timeout
+            )
+        except salobj.AckTimeoutError:
+            self.log.exception("Timed out waiting for the command ack. Continuing.")
+
+        # Make sure that a new lfo was created
+        lfo = await self.electrometer.evt_largeFileObjectAvailable.next(
+            timeout=self.long_timeout, flush=False
+        )
+        return lfo
+
+    async def take_fiber_spectrum(
+        self,
+        delay: float,
+        integration_time: float,
+    ) -> salobj.type_hints.BaseMsgType:
+        """Wait, then start an acquisition with the fiber spectrograph.
+
+        By default, this method will wait for `delay` seconds then start
+        an acquisition with the fiber spectrograph. Optionally the user may
+        provide a coroutine that will be awaited before the delay starts.
+
+        Parameters
+        ----------
+        delay : `float`
+            Seconds to wait before starting fiber spectrograph acquisition.
+        integration_time : `float`
+            Integration time for the fiber spectrum (seconds).
+
+        Returns
+        -------
+        RuntimeError : If exposure or return of lfo doesn't occur.
+        """
+        self.fiberspectrograph.evt_largeFileObjectAvailable.flush()
+
+        try:
+            await self.fiberspectrograph.cmd_expose.set_start(
+                integrationTime=integration_time,
+                timeout=integration_time + self.long_timeout,
+            )
+        except salobj.AckTimeoutError:
+            self.log.exception("Timed out waiting for the command ack. Continuing.")
+
+        lfo = await self.fiberspectrograph.evt_largeFileObjectAvailable.next(
+            timeout=self.long_timeout, flush=False
+        )
+        return lfo
+
+    async def perform_flat(self, exptime_dict: dict[str, float]) -> None:
+        # check dictionary
+        # Get the image number
+        # Somehow need to make sure camera is setup
+        assert self.latiss is not None
+        asyncio.gather(
+            self.electrometer_scan(duration=exptime_dict["electrometer"]),
+            self.take_fiber_spectrum(
+                delay=self.delay, integration_time=exptime_dict["fiberspectrograph"]
+            ),
+            self.latiss.take_flats(exptime=exptime_dict["camera"], nflats=1),
+        )
+
+    async def get_optimized_exposure_times(
+        self,
+        calib_type: typing.Optional[CalibrationType],
+        wavelength: typing.Optional[float],
+        **kwargs: typing.Union[int, float, str, None],
+    ) -> dict[str, float]:
+        """Need exposure time for camera, electrometer and fiberspectrograph"""
+        self.check_kwargs(**kwargs)
+
+        latiss_exptime = 10
+        electrometer_exptime = fs_exptime = latiss_exptime
+        self.exptime_dict["camera"] = float(latiss_exptime)
```

Comment:
This is a bit confusing. Why are you updating the value of the class attribute dictionary and then returning it? 

I think just create a new dictionary and return it.

I imagine this is a temporary implementation, so just do:

```
return dict(
    camera = 10,
    electrometer = 10,
    fiberspectrograph = 10,
)
```

and maybe create a ticket for the actual implementation and add a TODO here.

Reply:
done. https://rubinobs.atlassian.net/browse/DM-44361